resolvers.js

```diff
@@ -509,18 +538,50 @@ const Resolvers = {
         // if internal network is used.
         //
         // TODO support image storage running on a separate host
-        imageUrl = 'http://localhost:' + config.img_storage_port + imageUrl;
+        imageUrl = basePath + imageUrl;
       }
       const payload = jwt.decode(input.jwt, config.jwt.secret);
       try {
-        logger.info(input);
         await checkPermissions(datasetId, payload);
         const url = `http://${config.services.sm_engine_api_host}/v1/datasets/${datasetId}/add-optical-image`;
         const body = {url: imageUrl, transform};
-        await fetch(url, {
+        let processOptImage = await fetch(url, {
           method: 'POST',
           body: JSON.stringify(body),
           headers: {'Content-Type': 'application/json'}});
+        await checkFetchRes(processOptImage);
+        return 'success';
+      } catch (e) {
+        logger.error(e.message);
+        return e.message;
+      }
+    },
+
+    async deleteOpticalImage(_, args) {
+      let {datasetId} = args;
+      let allUrls = [];
+      const payload = jwt.decode(args.jwt, config.jwt.secret);
+      const basePath = `http://localhost:${config.img_storage_port}`;
+      const smAPIUrl = `http://${config.services.sm_engine_api_host}/v1/datasets/${datasetId}/del-optical-image`;
+      try {
+        await checkPermissions(datasetId, payload);
+
+        let rawOpticalImage = await pg.select('optical_image').from('dataset').where('id', '=', datasetId);
+        let opticalImageIds = await pg.select('id').from('optical_image').where('ds_id', '=', datasetId);
+        await allUrls.push(basePath +
+            `${config.img_upload.categories.raw_optical_image.path}delete/${rawOpticalImage[0].optical_image}`);
+        for (let image of opticalImageIds) {
+          await allUrls.push(basePath + `${config.img_upload.categories.optical_image.path}delete/${image.id}`)}
```

Comment:
It seems `push()` is synchronous, so I'd remove `await` from this line and the one above. Also please put the closing brace to a new line :)